examples/text-generation/utils.py

```diff
@@ -373,15 +373,22 @@ def initialize_model(args, logger):
         "revision": args.model_revision,
         "token": args.token,
     }
+    if args.disk_offload:
+        model_kwargs["device_map"] = "auto"
+        model_kwargs["offload_folder"] = "/tmp/offload_folder/"
+
     model = (
         setup_model(args, model_dtype, model_kwargs, logger)
         if not use_deepspeed
         else setup_distributed_model(args, model_dtype, model_kwargs, logger)
     )
     tokenizer, model = setup_tokenizer(args, model)
     generation_config = setup_generation_config(args, model, tokenizer)
+
+    if args.const_serialization_path:
+        setup_const_serialization(args.const_serialization_path)
     if args.fp8:
-        model = setup_quantization(args, model)
+        model = setup_inference(args, model)
```

Comment:
why can't we directly use htcore.hpu_initialize(model) ?

Reply:
We want to do it organized with prints and all

Reply:
also the setup_inference is confusing as this util is for text-generation inference. can you move the import print and htcore.hpu_initialize(model) here?